### DISCUSSION TITLE
What is "SQ code"? Where can I find reference material for it?

### DISCUSSION THREAD

**Discussion by jonemo** [mchehab/zbar#232] [General]
Support for a barcode symbology "SQ code" was added to zbar in 2019 in https://github.com/mchehab/zbar/pull/2 by @jasp00. Can anyone share a link or other reference for where I can find out more about this 2D symbology? I am particularly interested in the official specification of the code as well as any licensing information related to its use. 

What I was able to find so far:

I tried to find barcodes similar to the examples in zbar ([1] and [2]) in various online and printed references on barcodes and couldn't find anything visually similar. I also tried two commercial barcode decoding softwares with broad symbology support and they failed to decode the examples.

Web searches led me to [3], [4], and [5]. Maybe there is something to be discovered there, but my lack of Italian language skills and the many dead links made me give up after some searching.

[1]: https://github.com/mchehab/zbar/blob/master/examples/sqcode1-generated.png
[2]: https://github.com/mchehab/zbar/blob/master/examples/sqcode1-scanned.png
[3]: https://www.gt50.org/products/sqcode.php
[4]: https://play.google.com/store/apps/details?id=it.gt50.universalqreader
[5]: https://www.timbrodigitale.com

**Comment by jasp00** [comment]
Besides the decoding code in ZBar, I am virtually your only source about SQ code. Feel free to ask if you need more information, but please be patient, I may reply after a week.

SQ code is a patent-free technology. Barcode layout and wrapper format are public domain. I coded a wrapper and an encoder to XPM. These two programs are under GPLv3+, but they are simple; it would be easy to start from scratch in order to use a different license. I can share the source of these programs with you.

**Comment by jonemo** [comment]
Thank you for the reply! Do SQ codes exist "in the wild"? Are there places someone could go or documents to read where an SQ code would be found?

**Reply by jasp00** [reply]
No organization requires SQ code as far as I know. SQ code was designed to add printed digital signatures, so you could find such documents that were submitted to a governmental registry; I can show you examples of my documents. As a generic file container, you may find SQ codes in file transfers by paper, printed backups, and websites that only host images with lossy format.

**Comment by les-citrons** [comment]
hello, I would like to know more. who invented the SQ code?

**Reply by jasp00** [reply]
The idea of a simple 2D bar code may come from a lesson about bar codes at university, though I am not sure. The author of the current implementation known as SQ code is Javier Serrano Polo.
